app/controllers/admin/api_controller.rb

```diff
@@ -5,18 +5,29 @@ class Admin::ApiController < Admin::AdminController
   # If we used "api_key", then our user provider would try to use the value for authentication
 
   def index
-    keys = ApiKey
-      .where(hidden: false)
-      .includes(:user, :api_key_scopes)
+    offset = (params[:offset] || 0).to_i
+    limit = (params[:limit] || 50).to_i.clamp(1, 50)
 
-    # Put active keys first
-    # Sort active keys by created_at, sort revoked keys by revoked_at
-    keys = keys.order(<<~SQL)
-      CASE WHEN revoked_at IS NULL THEN 0 ELSE 1 END,
-      COALESCE(revoked_at, created_at) DESC
-    SQL
+    keys = ApiKey.where(hidden: false)
+    count = keys.count
 
-    render_serialized(keys.to_a, ApiKeySerializer, root: 'keys')
+    keys = keys
+      .includes(:user, :api_key_scopes)
+      # Put active keys first
+      # Sort active keys by created_at, sort revoked keys by revoked_at
+      .order(<<~SQL)
+        CASE WHEN revoked_at IS NULL THEN 0 ELSE 1 END,
+        COALESCE(revoked_at, created_at) DESC
+      SQL
+      .offset(offset)
+      .limit(limit)
+
+    render_json_dump(
+      keys: serialize_data(keys, ApiKeySerializer),
+      total_count: count,
```

Comment:
Just want to confirm this field is being used by the client because I did a quick scan and it doesn't seem like the client consumes this field.

Reply:
I wanted to use it, but I removed it then.